hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+    # --------------------------------------------
+
+    def get_tips(self) -> Deferred[list[bytes]]:
+        """Async method to request the tips, returned hashes guaranteed to be new"""
+        key = 'tips'
+        deferred = self.deferred_by_key.get(key, None)
+        if deferred is None:
+            deferred = self.deferred_by_key[key] = Deferred()
+            self.send_get_tips()
+        else:
+            assert self._receiving_tips is not None
+        return deferred
+
+    def send_get_tips(self) -> None:
+        self.log.debug('get tips')
+        self.send_message(ProtocolMessages.GET_TIPS)
+        self._receiving_tips = []
+
+    def handle_get_tips(self, payload: str) -> None:
+        """Handle a received GET_TIPS message."""
+        assert self.tx_storage.indexes is not None
+        assert self.tx_storage.indexes.mempool_tips is not None
+        if self._is_streaming:
+            self.log.warn('can\'t send while streaming')  # XXX: or can we?
+            self.send_message(ProtocolMessages.MEMPOOL_END)
+            return
+        self.log.debug('handle_get_tips')
+        # TODO Use a streaming of tips
+        for txid in self.tx_storage.indexes.mempool_tips.get():
+            self.send_tips(txid)
+        self.send_message(ProtocolMessages.TIPS_END)
+
+    def send_tips(self, tx_id: bytes) -> None:
+        """Send a TIPS message."""
+        self.send_message(ProtocolMessages.TIPS, json.dumps([tx_id.hex()]))
+
+    def handle_tips(self, payload: str) -> None:
+        """Handle a received TIPS message."""
+        self.log.debug('tips', receiving_tips=self._receiving_tips)
+        if self._receiving_tips is None:
+            self.protocol.send_error_and_close_connection('TIPS not expected')
+            return
+        data = json.loads(payload)
+        data = [bytes.fromhex(x) for x in data]
+        # filter-out txs we already have
+        self._receiving_tips.extend(tx_id for tx_id in data if not self.tx_storage.transaction_exists(tx_id))
+
+    def handle_tips_end(self, payload: str) -> None:
+        """Handle a received TIPS-END message."""
+        assert self._receiving_tips is not None
+        key = 'tips'
+        deferred = self.deferred_by_key.pop(key, None)
+        if deferred is None:
+            self.protocol.send_error_and_close_connection('TIPS-END not expected')
+            return
+        deferred.callback(self._receiving_tips)
+        self._receiving_tips = None
+
+    # ------------------------------------------
+    # END: GET_TIPS/TIPS/TIPS_END implementation
+    # ------------------------------------------
+
+    def send_relay(self) -> None:
+        self.log.debug('ask for relay')
+        self.send_message(ProtocolMessages.RELAY)
+
+    def handle_relay(self, payload: str) -> None:
+        """Handle a received RELAY message."""
+        # XXX: we need a way to turn this off, should we have arguments like: OFF, ALWAYS, SYNCED, ...? there is no
```

Comment:
I agree with the comment. Let's create an issue to get this done soon.

Reply:
It's now implemented in this PR.

Reply:
Thanks!